### ISSUE TITLE
how to train casanovo v4 on huge dataset like massive-kb from scratch?

### ISSUE THREAD

**Issue by irleader** [Noble-Lab/casanovo#314]
Hi,

I remeber that you mentioned you trained casanovo on massive-kb dataset with 1 epoch only, which gets the massive-kb pre-trained model.

1. If I also want to train on a huge dataset like massive-kb from scratch (I only need a subset of modifcations from massive-kb), is it also best to train only 1 epoch?

2. What is your suggestion to adjust the config.yaml parameters to train this 1 epoch? I assume it should be quite different from current setting for training multiple epoches until convergence.

   e.g.  
  warmup_iters: 100_000, 
  max_iters: 600_000,
  train_batch_size: 32,
  learning_rate: 5e-4, 
  weight_decay: 1e-5, 
  train_label_smoothing: 0.01
  
    What is your strategy of training huge dataset on casanovo? Is it possible to provide the config.yaml for this strategy (or at least your values for the above 6 variable combinations)?

Thanks a lot in advance!

Best regards

**Comment by bittremieux** [comment]
1. Rather than a specific number of epochs (or iterations for such a large training dataset), I recommend that you determine how long to train based on a sequence-disjoint validation set. That's also what we did when training on MassIVE-KB, and based on the validation performance we empirically noticed that the model converged already prior to having gone through a single full epoch.
2. No, these are the settings that we consistently use to train the various models. No special change needed.

**Comment by irleader** [comment]
Thanks a lot for your advice!

If I would like to fine-tune your massivekb pretrained model for my specific data, I simply need to continue train casanovo with your model and my own data (my data might be slightly different from massivekb) until it performs best on my own validation set?

**Comment by wsnoble** [comment]
Yes, that's right.
